### PR TITLE
Adds support for the custom health checks for the new cert-manager API. Closes #2667

### LINKED ISSUE #2667: Support for the custom health checks for the new cert-manager API
# Summary

Argocd has custom health checks for Issuer and Certificate resource of cert-manager.
Starting with cert-manager v0.11.0, the API group and version of these CRDs are changed as shown [here](https://github.com/jetstack/cert-manager/releases).
Therefore, it would be nice to have custom health checks for the new cert-manager API.

# Motivation

To behave in the same way for cert-manager v0.11.0 or higher.

# Proposal

Create cert-manager.io directory in resource_customizations, copy the contents of cert-manager.k8s.io directory, then fix apiVersion.

### PR DESCRIPTION
**Pull request by binoue**
For #2667 , this pull request made the following changes:
- add custom health checks for the new cert-manager API
- replace old API with new API in documents and tests

Please review if you have time.

Checklist:
* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Optional. My organization is added to the README.
* [x] I've signed the CLA and my build is green ([troubleshooting builds](https://argoproj.github.io/argo-cd/developer-guide/ci/)). 

Closes #2667

### CODE REVIEW

File: docs/operator-manual/argocd-cm.yaml [hunk 1]
```diff
@@ -101,7 +101,7 @@ data:
       ignoreDifferences: |
         jsonPointers:
         - webhooks/0/clientConfig/caBundle
-    certmanager.k8s.io/Certificate:
```
Comment (alexec) [inline docs/operator-manual/argocd-cm.yaml]: don't we need both of these?

Comment (binoue) [reply docs/operator-manual/argocd-cm.yaml]: As a document, I thought it is enough to use only the latest cert-manager API.
Is it better to write some comments here or add `resource.customizations` 
for old and new cert-manager API? 🤔 

(This pull request supports the old and new cert-manager API.)

Comment (alexec) [reply docs/operator-manual/argocd-cm.yaml]: I don’t think we want to break backwards compatibility.

Comment (binoue) [reply docs/operator-manual/argocd-cm.yaml]: Thanks! I understood. 
I will fix this immediately.

**Review by alexec — COMMENTED** [review]
Ok. What we need is someone who knows Cert manager to test these out. What about if you ask in the #argo-cd Slack channel?

**Comment by alexec** [comment]
Does this fix #2239?

**Comment by binoue** [comment]
> Does this fix #2239?

Sorry for the late reply. This pull request does not fix #2239 
Better to include the ignoreDifferences made by @samvdb ?

The status field seems already removed in cert-manager v0.12.0

**Review by alexec — APPROVED** [review]

**Pull request merged**